okhttp-bom/build.gradle

```diff
@@ -0,0 +1,13 @@
+dependencies {
+  constraints {
+    api project(':mockwebserver')
```

Comment:
Can we do this in a loop? And maybe conditionally based on if the `maven-publish` plugin was applied?

```groovy
dependencies {
  constraints {
    rootProject.subprojects { subproject ->
      if (subproject != project) {
        subproject.plugins.withId('maven-publish') {
          api subproject
        }
      }
    }
  }
}
```

I have no idea if this works.

Reply:
Unfortunately it doesn’t work because this runs before all of the subprojects’ plugins are applied. But I got something almost as good that avoids duplication:

```
dependencies {
  constraints {
    rootProject.subprojects { subproject ->
      def artifactId = rootProject.ext.publishedArtifactId(subproject)
      if (artifactId != null && project.name != 'okhttp-bom') {
        api subproject
      }
    }
  }
}
```